curvefs/src/metaserver/transaction.cpp

```
@@ -29,84 +29,101 @@ namespace metaserver {
     os << "txId = " << renameTx.txId_;
     for (size_t i = 0; i < dentrys.size(); i++) {
-        os << ", dentry[" << i << "] = ("
-           << dentrys[i].ShortDebugString() << ")";
+        os << ", dentry[" << i << "] = (" << dentrys[i].ShortDebugString()
+           << ")";
     }
     return os;
 }
 
-TxManager::TxManager(std::shared_ptr<DentryStorage> storage)
-    : storage_(storage) {}
+TxManager::TxManager(std::shared_ptr<DentryStorage> storage,
+                     common::PartitionInfo partitionInfo)
+    : storage_(std::move(storage)), conv_(std::make_shared<Converter>()),
+      partitionInfo_(std::move(partitionInfo)) {}
 
-MetaStatusCode TxManager::PreCheck(const std::vector<Dentry>& dentrys) {
+MetaStatusCode TxManager::PreCheck(const std::vector<Dentry> &dentrys) {
     auto size = dentrys.size();
     if (size != 1 && size != 2) {
         return MetaStatusCode::PARAM_ERROR;
     } else if (size == 2) {
-         if (dentrys[0].fsid() != dentrys[1].fsid() ||
-             dentrys[0].txid() != dentrys[1].txid() ||
-             dentrys[0].txsequence() != dentrys[1].txsequence()) {
+        if (dentrys[0].fsid() != dentrys[1].fsid() ||
+            dentrys[0].txid() != dentrys[1].txid() ||
+            dentrys[0].txsequence() != dentrys[1].txsequence()) {
             return MetaStatusCode::PARAM_ERROR;
         }
     }
 
     return MetaStatusCode::OK;
 }
 
-MetaStatusCode TxManager::HandleRenameTx(const std::vector<Dentry>& dentrys) {
+void TxManager::SerializeRenameTx(const RenameTx &in,
+                                  PrepareRenameTxRequest *out) {
+    auto *dentrys = in.GetConstDentrys();
+    out->set_poolid(partitionInfo_.poolid());
+    out->set_copysetid(partitionInfo_.copysetid());
+    out->set_partitionid(partitionInfo_.partitionid());
+    *out->mutable_dentrys() = {dentrys->begin(), dentrys->end()};
+}
+
+void TxManager::LoadPengingTx() {
+    metaserver::TransactionRequest request;
+    if (storage_->GetPendingTx(&request)) {
```

`GetPendingTx` may return false, is it safe to ignore this error?

---

Yes, if the snapshot generated by old version of metaserver, there are no transaction in snapshot.

---

Or we may meet a disk error, in that case nothing we can do, and the snapshot files maybe broken.

---

I have expose error to `MetaStore::Load()`.